libutils/json-utils.c

```diff
@@ -286,17 +326,29 @@ bool JsonParseCsvFile(const char *input_path, size_t size_max, JsonElement **jso
     }
 
     bool reached_eof = feof(fin);
-    fclose(fin);
 
     if (!reached_eof && byte_count <= size_max)
     {
         Log(LOG_LEVEL_ERR,
             "%s: unable to read line from CSV file '%s'. (fread: %s)",
             myname, input_path, GetErrorStr());
         JsonDestroy(json);
+        fclose(fin);
         return false;
     }
+    else if (JsonLength(json) == 0)
+    {
+        Log(LOG_LEVEL_WARNING,
+            "%s: CSV file '%s' is not empty, but nothing was parsed",
+            myname, input_path);
+        if (contains_newline(fin))
```

Comment:
I don't think this is really necessary. I think the warning should be printed for all non-empty files which don't parse to any data.

Reply:
Alright, I'll scrap the function then.